File: core/01-intro-gates/01_addTwoDigits.py
"""
You are given a two-digit integer n. Return the sum of its digits.

Example

For n = 29, the output should be
solution(n) = 11
"""

def solution(n):
    print()
    print(n)
    return sum(int(d) for d in str(n))


def main():
    data = 29
    r = solution(data)
    sol = 11
    print(f"{sol} -> {r} \t {sol == r}")


def test():
    pass


if __name__ == '__main__':
    main()
    # test()

File: src/core/01-intro-gates/01_addTwoDigits.rs
/*
You are given a two-digit integer n. Return the sum of its digits.

Example

For n = 29, the output should be
solution(n) = 11
*/

fn solution(n: i32) -> i32 {
    let mut sum = 0;
    let str_n = n.to_string();
    for c in str_n.chars() {
        sum += c.to_digit(10).unwrap();
    }
    sum as i32
}


fn main() {
    let data = 29;
    let r = solution(data);
    let sol = 11;
    println!("{} -> {} \t {}", sol, r, sol == r);
}
